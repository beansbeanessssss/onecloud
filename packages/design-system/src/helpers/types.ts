export type IconFillType = 'fill' | 'line' | 'none'
export type IconType = {
  name: string
  color?: string
  fillType?: IconFillType
}
