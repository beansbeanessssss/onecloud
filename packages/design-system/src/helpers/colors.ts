/**
 * Converts hex to rgb
 * @param {string} hex: The hex color value (with or without #)
 * @return {Array<number>|null} The converted rgb array
 **/
export function hexToRgb(hex: string): Array<number> | null {
  hex = hex.startsWith('#') ? hex : `#${hex}`
  let result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
  return result ? [parseInt(result[1], 16), parseInt(result[2], 16), parseInt(result[3], 16)] : null
}

/**
 * Converts rgb to hex
 * @param {Array<number>} rgbArray: The rgb color
 * @return {string} The converted hex value
 **/
export function rgbToHex(rgbArray: Array<number>): string {
  const [r, g, b] = rgbArray
  const rHex = r.toString(16).padStart(2, '0')
  const gHex = g.toString(16).padStart(2, '0')
  const bHex = b.toString(16).padStart(2, '0')
  return `#${rHex}${gHex}${bHex}`
}

function shadeValue(value: number, percent: number): string {
  let a = (value * (100 + percent)) / 100
  a = a < 255 ? a : 255
  a = Math.round(a)
  return a.toString(16).length == 1 ? '0' + a.toString(16) : a.toString(16)
}

/**
 * Dim or brighten a hex color
 * @param {string} rgb: The rgb color value
 * @return {string} The brightened or dimmed hex color
 **/
export function shadeColor(rgb: Array<number>, percent: number): string {
  let r = rgb[0]
  let g = rgb[1]
  let b = rgb[2]

  return `#${shadeValue(r, percent)}${shadeValue(g, percent)}${shadeValue(b, percent)}`
}

/**
 * Get the luminance of an rgb color
 * @param {Array<number>} rgb: The rgb value as an array
 * @return {Number} Returns value between 0 and 1, where 1 is white
 **/
export function getLuminanace(rgb: Array<number>): number {
  const mappedRgb = rgb.map((v) => {
    const val = v / 255
    return val <= 0.03928 ? val / 12.92 : ((val + 0.055) / 1.055) ** 2.4
  })
  return Number((0.2126 * mappedRgb[0] + 0.7152 * mappedRgb[1] + 0.0722 * mappedRgb[2]).toFixed(3))
}

/**
 * Get the contrast ratio between two rgb colors
 * @param {Array<number>} rgbColorA: The first rgb value as an array
 * @param {Array<number>} rgbColorB: The second rgb value as an array
 * @return {Number} Returns value between 1 and 21, where 1 is no contrast and 21 is max contrast
 **/
export function getContrastRatio(rgbColorA: Array<number>, rgbColorB: Array<number>): number {
  const lumA = getLuminanace(rgbColorA)
  const lumB = getLuminanace(rgbColorB)

  return (Math.max(lumA, lumB) + 0.05) / (Math.min(lumA, lumB) + 0.05)
}

/**
 * Gives you a random hashed color for a string, e.g. if you give it 'owncloud' it will always return the same color
 * @param {string} name: Can be any string
 * @return {string} Returns a hex color
 **/
export function generateHashedColorForString(name: string): string {
  let hash = 0
  for (let i = 0; i < name.length; i++) {
    hash = name.charCodeAt(i) + ((hash << 5) - hash)
  }
  return `#${(hash & 0x00ffffff).toString(16).toUpperCase()}`
}

/**
 * Adjusts a given color to match the contrast ratio of another color
 * @param {Array<number>} targetColorRgb: color to adjust
 * @param {Array<number>} associatedColorRgb: brightest reference color
 * @param {Array<number>} desiredRatio: desired contrast ratio
 * @return {string} Returns a rgb color array
 **/
export function setDesiredContrastRatio(
  targetColorRgb: Array<number>,
  associatedColorRgb: Array<number>,
  desiredRatio: number,
  dim = true
): Array<number> {
  const ratio = getContrastRatio(targetColorRgb, associatedColorRgb)
  if (Math.abs(desiredRatio - ratio) <= 0.3) {
    return targetColorRgb
  }
  if (ratio < desiredRatio) {
    return setDesiredContrastRatio(
      hexToRgb(shadeColor(targetColorRgb, -1)),
      associatedColorRgb,
      desiredRatio,
      true
    )
  }
  return setDesiredContrastRatio(
    hexToRgb(shadeColor(targetColorRgb, 1)),
    associatedColorRgb,
    desiredRatio,
    false
  )
}
