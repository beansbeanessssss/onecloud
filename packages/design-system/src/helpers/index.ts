export * from './constants'
export * from './types'
export * from './colors'
