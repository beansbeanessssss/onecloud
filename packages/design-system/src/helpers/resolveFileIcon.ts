const archiveExtensions = [
  '7z',
  'apk',
  'bz2',
  'deb',
  'gz',
  'gzip',
  'rar',
  'tar',
  'tar.bz2',
  'tar.gz',
  'tbz2',
  'tgz',
  'zip'
]

const audioExtensions = [
  '3gp',
  '8svx',
  'aa',
  'aac',
  'aax',
  'act',
  'aiff',
  'alac',
  'amr',
  'ape',
  'au',
  'awb',
  'cda',
  'dss',
  'dvf',
  'flac',
  'gsm',
  'iklax',
  'ivs',
  'm4a',
  'm4b',
  'm4p',
  'mmf',
  'mogg',
  'movpkg',
  'mp3',
  'mpc',
  'msv',
  'nmf',
  'oga',
  'ogga',
  'opus',
  'ra',
  'raw',
  'rf64',
  'rm',
  'sln',
  'tta',
  'voc',
  'vox',
  'wav',
  'wma',
  'wv'
]

const codeExtensions = [
  'bash',
  'c++',
  'c',
  'cc',
  'cpp',
  'css',
  'feature',
  'go',
  'h',
  'hh',
  'hpp',
  'htm',
  'html',
  'java',
  'js',
  'json',
  'php',
  'pl',
  'py',
  'scss',
  'sh',
  'sh-lib',
  'sql',
  'ts',
  'xml',
  'yaml',
  'yml'
]

const defaultExtensions = ['accdb', 'rss', 'swf']

const documentExtensions = ['doc', 'docm', 'docx', 'dot', 'dotx', 'lwp', 'odt', 'one', 'vsd', 'wpd']

const formExtensions = ['docf', 'docxf', 'oform']

const imageExtensions = [
  'ai',
  'cdr',
  'eot',
  'eps',
  'gif',
  'jpeg',
  'jpg',
  'otf',
  'pfb',
  'png',
  'ps',
  'psd',
  'svg',
  'ttf',
  'woff',
  'xcf'
]

const presentationExtensions = [
  'odp',
  'pot',
  'potm',
  'potx',
  'ppa',
  'ppam',
  'pps',
  'ppsm',
  'ppsx',
  'ppt',
  'pptm',
  'pptx'
]

const spreadsheetExtensions = [
  'csv',
  'ods',
  'xla',
  'xlam',
  'xls',
  'xlsb',
  'xlsm',
  'xlsx',
  'xlt',
  'xltm',
  'xltx'
]

const textExtensions = [
  'cb7',
  'cba',
  'cbr',
  'cbt',
  'cbtc',
  'cbz',
  'cvbdl',
  'eml',
  'epub',
  'mdb',
  'tex',
  'txt'
]

const videoExtensions = ['mov', 'mp4', 'webm', 'wmv']

const fileIcon = {
  archive: { name: 'resource-type-archive', color: 'var(--oc-color-icon-archive)' },
  audio: { name: 'resource-type-audio', color: 'var(--oc-color-icon-audio)' },
  code: { name: 'resource-type-code', color: 'var(--oc-color-text-default)' },
  default: { name: 'resource-type-file', color: 'var(--oc-color-text-default)' },
  drawio: { name: 'resource-type-drawio', color: 'var(--oc-color-icon-drawio)' },
  document: { name: 'resource-type-document', color: 'var(--oc-color-icon-document)' },
  ifc: { name: 'resource-type-ifc', color: 'var(--oc-color-icon-ifc)' },
  ipynb: { name: 'resource-type-jupyter', color: 'var(--oc-color-icon-jupyter)' },
  image: { name: 'resource-type-image', color: 'var(--oc-color-icon-image)' },
  form: { name: 'resource-type-form', color: 'var(--oc-color-icon-form)' },
  markdown: { name: 'resource-type-markdown', color: 'var(--oc-color-icon-markdown)' },
  odg: { name: 'resource-type-graphic', color: 'var(--oc-color-icon-graphic)' },
  pdf: { name: 'resource-type-pdf', color: 'var(--oc-color-icon-pdf)' },
  presentation: { name: 'resource-type-presentation', color: 'var(--oc-color-icon-presentation)' },
  root: { name: 'resource-type-root', color: 'var(--oc-color-icon-root)' },
  spreadsheet: { name: 'resource-type-spreadsheet', color: 'var(--oc-color-icon-spreadsheet)' },
  text: { name: 'resource-type-text', color: 'var(--oc-color-text-default)' },
  url: { name: 'resource-type-url', color: 'var(--oc-color-text-default)' },
  video: { name: 'resource-type-video', color: 'var(--oc-color-icon-video)' }
}

export function resolveFileIcon(fileExtension: string) {
  if (archiveExtensions.includes(fileExtension)) {
    return fileIcon.archive
  }
  if (audioExtensions.includes(fileExtension)) {
    return fileIcon.audio
  }
  if (codeExtensions.includes(fileExtension)) {
    return fileIcon.code
  }
  if (defaultExtensions.includes(fileExtension)) {
    return fileIcon.default
  }
  if (documentExtensions.includes(fileExtension)) {
    return fileIcon.document
  }
  if (imageExtensions.includes(fileExtension)) {
    return fileIcon.image
  }
  if (formExtensions.includes(fileExtension)) {
    return fileIcon.form
  }
  if (presentationExtensions.includes(fileExtension)) {
    return fileIcon.presentation
  }
  if (spreadsheetExtensions.includes(fileExtension)) {
    return fileIcon.spreadsheet
  }
  if (textExtensions.includes(fileExtension)) {
    return fileIcon.text
  }
  if (videoExtensions.includes(fileExtension)) {
    return fileIcon.video
  }

  if (fileExtension === 'drawio') {
    return fileIcon.drawio
  }
  if (fileExtension === 'ifc') {
    return fileIcon.ifc
  }
  if (fileExtension === 'ipynb') {
    return fileIcon.ipynb
  }
  if (fileExtension === 'md') {
    return fileIcon.markdown
  }
  if (fileExtension === 'odg') {
    return fileIcon.odg
  }
  if (fileExtension === 'pdf') {
    return fileIcon.pdf
  }
  if (fileExtension === 'root') {
    return fileIcon.root
  }
  if (fileExtension === 'url') {
    return fileIcon.url
  }

  return fileIcon.default
}
