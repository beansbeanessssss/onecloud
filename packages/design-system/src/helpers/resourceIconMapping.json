{
  "7z": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "accdb": {
    "name": "resource-type-file",
    "color": "var(--oc-color-text-default)"
  },
  "ai": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "apk": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "bash": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "bz2": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "c++": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "c": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "cb7": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cba": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cbr": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cbt": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cbtc": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cbz": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "cc": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "cdr": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "cpp": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "css": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "csv": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "cvbdl": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "dcm": {
    "name": "resource-type-medical",
    "color": "var(--oc-color-icon-medical)"
  },
  "deb": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "doc": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "docm": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "docx": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "dot": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "dotx": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "drawio": {
    "name": "resource-type-drawio",
    "color": "var(--oc-color-icon-drawio)"
  },
  "eml": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "eot": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "eps": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "epub": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "feature": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "go": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "gif": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "gz": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "gzip": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "h": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "hh": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "hpp": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "htm": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "html": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "ifc": {
    "name": "resource-type-ifc",
    "color": "var(--oc-color-icon-ifc)"
  },
  "ipynb": {
    "name": "resource-type-jupyter",
    "color": "var(--oc-color-icon-jupyter)"
  },
  "java": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "jpeg": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "jpg": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "js": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "json": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "lwp": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "md": {
    "name": "resource-type-file",
    "color": "var(--oc-color-text-default)"
  },
  "mdb": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "mov": {
    "name": "resource-type-video",
    "color": "var(--oc-color-icon-video)"
  },
  "mp3": {
    "name": "resource-type-audio",
    "color": "var(--oc-color-icon-audio)"
  },
  "mp4": {
    "name": "resource-type-video",
    "color": "var(--oc-color-icon-video)"
  },
  "odp": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "odg": {
    "name": "resource-type-graphic",
    "color": "var(--oc-color-icon-graphic)"
  },
  "ods": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "odt": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "one": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "otf": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "pdf": {
    "name": "resource-type-pdf",
    "color": "var(--oc-color-icon-pdf)"
  },
  "pfb": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "php": {
    "name": "resource-type-code",
    "color": "var(--oc-color-icon-default)"
  },
  "pl": {
    "name": "resource-type-code",
    "color": "var(--oc-color-icon-default)"
  },
  "png": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "pot": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "potm": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "potx": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ppa": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ppam": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "pps": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ppsm": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ppsx": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ppt": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "pptm": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "pptx": {
    "name": "resource-type-presentation",
    "color": "var(--oc-color-icon-presentation)"
  },
  "ps": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "psd": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "py": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "rar": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "root": {
    "name": "resource-type-root",
    "color": "var(--oc-color-icon-root)"
  },
  "rss": {
    "name": "resource-type-file",
    "color": "var(--oc-color-text-default)"
  },
  "scss": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "sh": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "sh-lib": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "sql": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "svg": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "swf": {
    "name": "resource-type-file",
    "color": "var(--oc-color-text-default)"
  },
  "tar": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "tar.bz2": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "tar.gz": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "tbz2": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "tex": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "tgz": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  },
  "ts": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "ttf": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "txt": {
    "name": "resource-type-text",
    "color": "var(--oc-color-text-default)"
  },
  "vsd": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "wav": {
    "name": "resource-type-audio",
    "color": "var(--oc-color-icon-audio)"
  },
  "webm": {
    "name": "resource-type-video",
    "color": "var(--oc-color-icon-video)"
  },
  "wmv": {
    "name": "resource-type-video",
    "color": "var(--oc-color-icon-video)"
  },
  "woff": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "wpd": {
    "name": "resource-type-document",
    "color": "var(--oc-color-icon-document)"
  },
  "xcf": {
    "name": "resource-type-image",
    "color": "var(--oc-color-icon-image)"
  },
  "xla": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xlam": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xls": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xlsb": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xlsm": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xlsx": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xlt": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xltm": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xltx": {
    "name": "resource-type-spreadsheet",
    "color": "var(--oc-color-icon-spreadsheet)"
  },
  "xml": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "yaml": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "yml": {
    "name": "resource-type-code",
    "color": "var(--oc-color-text-default)"
  },
  "zip": {
    "name": "resource-type-archive",
    "color": "var(--oc-color-icon-archive)"
  }
}
