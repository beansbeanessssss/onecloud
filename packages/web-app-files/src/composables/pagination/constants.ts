export abstract class PaginationConstants {
  static readonly perPageDefault: string = '100'
  static readonly perPageQueryName: string = 'items-per-page'
  static readonly pageName: string = 'page'
  static readonly pageNameDefault: string = '1'
}
