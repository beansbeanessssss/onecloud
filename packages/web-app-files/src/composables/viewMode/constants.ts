export abstract class ViewModeConstants {
  static readonly default: string = 'resource-table'
  static readonly queryName: string = 'view-mode'
  static readonly condensedTable: string = 'resource-table-condensed'
}
