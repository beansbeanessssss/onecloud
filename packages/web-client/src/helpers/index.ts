export * from './group'
export * from './item'
export * from './publicLink'
export * from './resource'
export * from './share'
export * from './space'
export * from './user'
