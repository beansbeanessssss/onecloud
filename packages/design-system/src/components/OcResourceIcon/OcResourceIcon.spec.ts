import { shallowMount } from '@vue/test-utils'

import { AVAILABLE_SIZES } from '../../helpers/constants'
import { OcResourceIcon } from '..'

const resources = ['file', 'folder', 'space']

function getWrapper(props) {
  return shallowMount(OcResourceIcon, {
    propsData: {
      resource: { type: props.resourceType },
      size: props.size
    }
  })
}

describe('OcResourceIcon', () => {
  resources.forEach((resourceType) => {
    AVAILABLE_SIZES.forEach((size) => {
      it(`renders OcIcon for resource type ${resourceType} in size ${size}`, () => {
        const wrapper = getWrapper({ resourceType, size })
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
