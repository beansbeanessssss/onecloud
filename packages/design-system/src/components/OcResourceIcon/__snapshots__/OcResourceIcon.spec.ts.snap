// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcResourceIcon renders OcIcon for resource type file in size large 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="large" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size medium 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size small 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size xlarge 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="xlarge" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size xsmall 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="xsmall" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size xxlarge 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="xxlarge" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type file in size xxxlarge 1`] = `<oc-icon-stub name="resource-type-file" filltype="fill" accessiblelabel="" type="span" size="xxxlarge" variation="passive" color="var(--oc-color-text-default)" class="oc-resource-icon oc-resource-icon-file"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size large 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="large" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size medium 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size small 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size xlarge 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="xlarge" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size xsmall 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="xsmall" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size xxlarge 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="xxlarge" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type folder in size xxxlarge 1`] = `<oc-icon-stub name="resource-type-folder" filltype="fill" accessiblelabel="" type="span" size="xxxlarge" variation="passive" color="var(--oc-color-icon-folder)" class="oc-resource-icon oc-resource-icon-folder"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size large 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="large" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size medium 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size small 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size xlarge 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="xlarge" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size xsmall 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="xsmall" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size xxlarge 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="xxlarge" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;

exports[`OcResourceIcon renders OcIcon for resource type space in size xxxlarge 1`] = `<oc-icon-stub name="layout-grid" filltype="fill" accessiblelabel="" type="span" size="xxxlarge" variation="passive" color="var(--oc-color-swatch-passive-default)" class="oc-resource-icon oc-resource-icon-space"></oc-icon-stub>`;
